src/qiboml/models/abstract.py

```diff
@@ -0,0 +1,40 @@
+"""Defines the general structure of a qiboml module"""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+from qibo import Circuit
+from qibo.config import raise_error
+from qibo.gates import abstract
+
+from qiboml.backends import JaxBackend
+
+
+@dataclass
+class QuantumCircuitLayer(ABC):
+
+    nqubits: int
+    qubits: list[int] = None
+    circuit: Circuit = None
+    initial_state: "ndarray" = None
+    backend: "qibo.backends.Backend" = JaxBackend()
```

Comment:
In this cases I am not sure whether it is worth importing objects just for type hinting. Here it should not be a problem, but I remember that in `qibo` I often encountered circular import errors.

Reply:
Yes, the circular imports are a problem. But the strings are not the solution (just a patch).

The main reason for allowing strings are mutual references. Which should be possible (valid at runtime) but not working.
There are many PEPs for postponed annotation, and it's the only feature that is still part of `__future__`, and delayed for 4.0.

In any case, I can give you examples of why the proper solution would be different when we'll encounter them. But it should never be the case for a `qibo` object in `qiboml` (since `qibo` should never import `qiboml`, not *statically*).